aeron-archive/src/main/java/io/aeron/archive/client/ReplayMerge.java

```diff
@@ -175,18 +176,23 @@ public void close()
         final State state = this.state;
         if (State.CLOSED != state)
         {
-            if (!archive.context().aeron().isClosed())
+            final AeronArchive.Context context = archive.context();
+            final Aeron aeron = context.aeron();
+            if (!aeron.isClosed())
             {
+                final ErrorHandler errorHandler = context.errorHandler();
                 if (State.MERGED != state && State.STOP_REPLAY != state)
                 {
-                    subscription.removeDestination(replayDestination);
+                    AeronCloseHelper.close(errorHandler, () -> subscription.removeDestination(replayDestination));
```

Comment:
This should be an asyncRemoveDestination and not need a close helper.

Reply:
Done